src/Rendering_Internals.md

```diff
@@ -0,0 +1,408 @@
+
+[^more_than_10]:
+Since these 10 object slots are only refilled during Mode 2, and there appears to be no way to manipulate the PPU mode in the middle of a scanline, it seems that this limitation cannot be worked around.
+
+## Pixel Slice Fetcher
+
+The fetcher grabs a row of 8 pixels at a time to be fed to either FIFO.
+Data is fetched from VRAM one byte at a time, thus pixels are always fetched 8 at a time[^fetch_size] (hence the FIFOs' width).
+
+The following steps are executed, in this order:
+
+1. [Get tile ID](<#Get tile ID>)
+1. [Get tile row (low)](<#Get tile row (low)>)
+1. [Get tile row (high)](<#Get tile row (high)>)
+1. [Push pixels](<#Push pixels>)
+
+[^fetch_size]:
+Well, since pixels are 2bpp, two fetches are necessary—but one still ends up with 8 pixels, since the VRAM data bus is 8-bit.
+
+### Get tile ID
+
+This step determines which background/window tile to fetch pixels from.
+This step is not executed by the Pixel Slice Fetcher itself, but rather by the active "fetcher":
+
+#### BG fetcher
+
+During this step, a tilemap is sampled to determine which tile to fetch.
+
+The address read depends on whether the BG fetcher is in ["BG mode" or "Window mode"](<#>):
+<table style="text-align: center;" class="compact">
+   <thead>
+      <tr>
+         <th>Mode</th><th>15</th><th>14</th><th>13</th><th>12</th><th>11</th><th>10</th><th>9</th><th>8</th><th>7</th><th>6</th><th>5</th><th>4</th><th>3</th><th>2</th><th>1</th><th>0</th>
+      </tr>
+   </thead>
+   <tbody>
+      <tr>
+         <th>BG</th><td rowspan="2">1</td><td rowspan="2">0</td><td rowspan="2">0</td><td rowspan="2">1</td><td rowspan="2">1</td><td>LCDC bit 3</td><td colspan="5">(LY + SCY) ∕ 8</td><td colspan="5">(LX + SCX) ∕ 8</td>
+      </tr>
+      <tr>
+         <th>Window</th><td>LCDC bit 6</td><td colspan="5">"Window Y" ∕ 8</td><td colspan="5">LX ∕ 8</td>
+      </tr>
+   </tbody>
+</table>
+
+::: tip Wrapping
+
+The two additions in the "BG mode" row are carried out in 8 bits, i.e. modulo 256.
+Due the division by 8, the modulo is essentially 32, i.e. a tilemap's width.
+This is what causes the background to wrap around, both vertically and horizontally.
+
+:::
+
+A byte is read from the computed address, and is forwarded to the Pixel Slice Fetcher as a tile ID.
+Color models read from the computed address twice: from VRAM bank 0 to get the tile ID, and from VRAM bank 1 to get the attributes.[^banks_parallel]
+
+::: tip Raster effects
+
+Interestingly, unlike e.g. the NES' PPU, great care has been taken to ensure that the BG fetcher re-reads as many registers as possible (`SCY`, `LCDC`, etc.).
+This may have been insight from the former console, on which [proper "raster splits" are quite tricky](https://www.nesdev.org/wiki/PPU_scrolling#Split_X_scroll) due to a lot of internal caching.
+
+:::
```

Comment:
This is interesting but might be a bit out of scope?
It sounds like trivia, due to the framing with the NES PPU.